/***************************************************************************
 *   Copyright (C) 2016-2021 Brocade Communications Systems, Inc.          *
 *   All rights reserved.                                                  *
 *                                                                         *
 *   Author: Brunda R Rajagopala                                           *
 *                                                                         *
 **************************************************************************/

syntax = "proto3";

/*
This module is an extension to telemetry model for
 - Defining RPCs for telemetry in the managed device.
*/

package telemetry_system_stats;


option optimize_for = CODE_SIZE;


service BrocadeTelemetrySystemData_service
{

    rpc BrocadeTelemetrySystemData (BrocadeTelemetrySystemData_request) returns (stream BrocadeTelemetrySystemData_response) {}

}

message BrocadeTelemetrySystemData_request
{
    string profile_name = 1;
}

message BrocadeTelemetrySystemData_response
{
    string profile_name = 1;
    uint32 total_system_memory = 2;
    uint32 total_used_memory = 3;
    uint32 total_free_memory = 4;
    uint32 cached_memory = 5;
    uint32 buffers = 6;
    uint32 user_free_memory = 7;
    uint32 kernel_free_memory = 8;
    uint32 total_swap_memory = 9;
    uint32 total_free_swap_memory = 10;
    uint32 total_used_swap_memory = 11;
    uint32 user_process = 12;
    uint32 system_process = 13;
    uint32 niced_process = 14;
    uint32 iowait = 15;
    uint32 hw_interrupt = 16;
    uint32 sw_interrupt = 17;
    uint32 idle_state = 18;
    uint32 steal_time=19;
    uint32 uptime=20;
}




